packages/eslint-plugin-wpcalypso/lib/rules/i18n-unlocalized-url.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file Disallow using unlocalized URL strings
+ * @author Automattic
+ * @copyright 2023 Automattic. All rights reserved.
+ * See LICENSE.md file in root directory for full license.
+ */
+
+//------------------------------------------------------------------------------
+// Constants
+//------------------------------------------------------------------------------
+
+const LOCALIZABLE_URLS = [
+	'apps.wordpress.com',
+	'automattic.com/cookies(/|$)',
+	'automattic.com/privacy(/|$)',
+	'wordpress.com/blog(/|$)',
+	'wordpress.com/forums(/|$)',
+	'wordpress.com/go(/|$)',
+	'wordpress.com/support(/|$)',
+	'wordpress.com/tos(/|$)',
+];
+
+//------------------------------------------------------------------------------
+// Helper Functions
+//------------------------------------------------------------------------------
+
+const getCallee = require( '../util/get-callee' );
+
+/**
+ * Check whether the provided string is localizable string.
+ *
+ * @param   {string}  string String to be tested
+ * @returns {boolean}
+ */
+function isLocalizableUrlString( string ) {
+	return LOCALIZABLE_URLS.some( ( url ) =>
+		new RegExp( `^(https?:)?//${ url.replace( '.', '\\.' ) }`, 'i' ).test( string )
```

Comment:
Perf nit: we create a new regex every time. I wonder if it's more straightforward to just create the regex objects inside `LOCALIZABLE_URLS` and use them directly. That's related to my comment above.